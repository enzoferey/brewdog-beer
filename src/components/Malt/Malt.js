import React, { Fragment } from "react";
import PropTypes from "prop-types";
import style from "./malt.scss";

import weigthIcon from "components/Table/weight.png";

import TableRowSection from "components/TableRowSection";
import IconLabel from "components/IconLabel";

const Malt = ({ name, amount: { value, unit } }) => (
  <Fragment>
    <TableRowSection>
      <span className={style.name}>{name}</span>
    </TableRowSection>
    <TableRowSection>
      <IconLabel icon={weigthIcon} label={`${value} ${unit}`} />
    </TableRowSection>
  </Fragment>
);

Malt.propTypes = {
  name: PropTypes.string,
  amount: PropTypes.shape({
    value: PropTypes.number,
    unit: PropTypes.string,
  }),
};

export default Malt;
