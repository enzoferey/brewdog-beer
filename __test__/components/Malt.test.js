import React from "react";
import Malt from "components/Malt";

const malt = mocks.beers[0].ingredients.malt[0];

test("renders correctly", () => {
  testSnapshot(<Malt {...malt} />);
});
