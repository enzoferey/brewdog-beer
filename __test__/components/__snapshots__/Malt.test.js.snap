// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className={undefined}
    style={
        Object {
            "flex": 1,
          }
    }
>
    <span
        className={undefined}
    >
        Maris Otter Extra Pale
    </span>
</div>,
  <div
    className={undefined}
    style={
        Object {
            "flex": 1,
          }
    }
>
    <img
        alt="Weight icon"
        className={undefined}
        src="weight.png"
    />
    <span
        className={undefined}
    >
        3.3 kilograms
    </span>
</div>,
]
`;
